reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -8955,7 +8964,25 @@ public final void subscribe(Subscriber<? super T> actual) {
 	 * @see #as as() for a loose conversion to an arbitrary type
 	 */
 	public final <V> Flux<V> transformDeferred(Function<? super Flux<T>, ? extends Publisher<V>> transformer) {
-		return defer(() -> transformer.apply(this));
+		return deferWithContext(ctx -> {
```

Comment:
we cannot really change this operator (we might want a separate one) because with a legit Reactor sub-sequence that writes the `Context` in the transform `Function`, we'd loose said writes :(